core/chains/evm/client/rpc_client.go

```diff
@@ -817,7 +821,51 @@ func (r *rpcClient) PendingCallContract(ctx context.Context, msg interface{}) (v
 	)
 
 	return
+}
+
+// COPIED FROM go-ethereum/ethclient/gethclient - must be kept up to date!
+func toBlockNumArg(number *big.Int) string {
+	if number == nil {
+		return "latest"
+	}
+	if number.Sign() >= 0 {
+		return hexutil.EncodeBig(number)
+	}
+	// It's negative.
+	if number.IsInt64() {
+		return rpc.BlockNumber(number.Int64()).String()
+	}
+	// It's negative and large, which is invalid.
+	return fmt.Sprintf("<invalid %d>", number)
+}
 
+// COPIED FROM go-ethereum/ethclient/gethclient - must be kept up to date!
+// Modified to include legacy 'data' as well as 'input' in order to support non-compliant servers.
+func toCallArg(msg ethereum.CallMsg) interface{} {
+	arg := map[string]interface{}{
+		"from": msg.From,
+		"to":   msg.To,
+	}
+	if len(msg.Data) > 0 {
+		arg["input"] = hexutil.Bytes(msg.Data)
+		arg["data"] = hexutil.Bytes(msg.Data) // duplicate legacy field for compatability
```

Comment:
This is interesting - so geth included these before and has now removed it in the latest version. Is that correct?

Reply:
This is indeed nonstandard: https://ethereum.org/en/developers/docs/apis/json-rpc#eth_call